src/main/scala/no/ndla/frontpageapi/service/ConverterService.scala

```diff
@@ -57,34 +59,44 @@ object ConverterService {
     filteredNames.map(name => api.MovieThemeName(name.name, name.language))
   }
 
-  def toApiSubjectPage(sub: domain.SubjectFrontPageData, language: String): api.SubjectPageData = {
-    api.SubjectPageData(
-      sub.id.get,
-      sub.name,
-      sub.filters,
-      sub.layout.toString,
-      sub.twitter,
-      sub.facebook,
-      toApiBannerImage(sub.bannerImage),
-      toApiAboutSubject(sub.about, language),
-      toApiMetaDescription(sub.metaDescription, language),
-      sub.topical,
-      sub.mostRead,
-      sub.editorsChoices,
-      sub.latestContent,
-      sub.goTo
-    )
+  def toApiSubjectPage(sub: domain.SubjectFrontPageData,
+                       language: String,
+                       fallback: Boolean = false): Try[api.SubjectPageData] = {
+    if (sub.supportedLanguages.contains(language) || fallback) {
+      Success(
+        api.SubjectPageData(
+          sub.id.get,
+          sub.name,
+          sub.filters,
+          sub.layout.toString,
+          sub.twitter,
+          sub.facebook,
+          toApiBannerImage(sub.bannerImage),
+          toApiAboutSubject(findByLanguageOrBestEffort(sub.about, language)),
+          toApiMetaDescription(findByLanguageOrBestEffort(sub.metaDescription, language)),
+          sub.topical,
+          sub.mostRead,
+          sub.editorsChoices,
+          sub.latestContent,
+          sub.goTo,
+          sub.supportedLanguages
+        )
+      )
```

Comment:
I praksis så vil aldri dette skje™, men dersom `sub.id` er `None` så vil `.get` kaste en exception.
Ser at det var sånn fra før av, men siden du først er her så kan vi sikkert endre denne til noe sånt som dette:
```suggestion
      sub.id match {
        case None => Failure(MissingIdException("Could not convert to api.SubjectPageData since domain object did not have an id"))
        case Some(subjectPageId) => 
          Success(
            api.SubjectPageData(
              subjectPageId,
              sub.name,
              sub.filters,
              sub.layout.toString,
              sub.twitter,
              sub.facebook,
              toApiBannerImage(sub.bannerImage),
              toApiAboutSubject(findByLanguageOrBestEffort(sub.about, language)),
              toApiMetaDescription(findByLanguageOrBestEffort(sub.metaDescription, language)),
              sub.topical,
              sub.mostRead,
              sub.editorsChoices,
              sub.latestContent,
              sub.goTo,
              sub.supportedLanguages
            )
          )
      }
```

Reply:
Skjønner! Hvorfor er id optional i domain.SubjectFrontPageData egentlig? Vil det ikke alltid lages en id for frontpagene?

Reply:
Tanken var nok at da kan vi lage en `domain.SubjectFrontPageData` som vi skal sette inn i databasen (før vi vet hva id'en blir). Kunne sikkert vært løst på flere måter :smile: